lib/events/multilog.go

```diff
@@ -26,17 +26,27 @@ import (
 )
 
 // NewMultiLog returns a new instance of a multi logger
-func NewMultiLog(loggers ...IAuditLog) *MultiLog {
-	return &MultiLog{
-		loggers: loggers,
+func NewMultiLog(loggers ...IAuditLog) (*MultiLog, error) {
+	emitters := make([]Emitter, len(loggers))
+	for i := range loggers {
+		emitter, ok := loggers[i].(Emitter)
+		if !ok {
+			return nil, trace.BadParameter("expected emitter, got %T", loggers[i])
+		}
+		emitters[i] = emitter
 	}
```

Comment:
nit: prefer using `append` when possible, it's harder to mess up indexes and go out of bounds
```suggestion
	emitters := make([]Emitter, 0, len(loggers))
	for _, l := range loggers {
		emitter, ok := l.(Emitter)
		if !ok {
			return nil, trace.BadParameter("expected emitter, got %T", l)
		}
		emitters = append(emitters, emitter)
	}
```